models/pull.go

```diff
@@ -287,6 +287,14 @@ func (pr *PullRequest) Merge(doer *User, baseGitRepo *git.Repository, mergeStyle
 	}
 	prConfig := prUnit.PullRequestsConfig()
 
+	if protected, err := pr.BaseRepo.IsProtectedBranch(pr.BaseBranch, doer); err != nil {
```

Comment:
Would be better to create new function `pr.CheckUserAllowedToMerge` that would return `error` (`nil` for allowed to merge)